geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaFeatureSource.scala

```diff
@@ -59,7 +60,12 @@ class GeoMesaFeatureSource(val ds: DataStore with HasGeoMesaStats,
     import org.locationtech.geomesa.index.conf.QueryProperties.QueryExactCount
 
     val useExactCount = query.getHints.isExactCount.getOrElse(QueryExactCount.get.toBoolean)
-    val count = ds.stats.getCount(getSchema, query.getFilter, useExactCount).getOrElse(-1L)
+
+    val count = if (query.getMaxFeatures < 1000) {
```

Comment:
maybe make the threshold a sys prop?

Reply:
Either a system property or just change this to `!query.isMaxFeaturesUnlimited`.

Thoughts?

Reply:
i'd go with sys prop, just to avoid regression with someone asking for a large but not unlimited number of features

Reply:
geomesa.exact.count.max.features - When EXACT_COUNT=true and maxFeatures is lower than this setting, GeoMesa will run the query to determine how many records are in the result set.  Otherwise, GeoMesa will use the Stats API and respect the EXACT_COUNT setting.